tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_min_full_scan_interval.py

```diff
@@ -0,0 +1,150 @@
+              f"'e7dbc9bba5a0ee252866536225b952d3de7ea5cb','777fef8cc434b597769d102361af718d29ef72c1')")
+    query_wdb(f"agent {agent.id} sql UPDATE sync_info SET last_attempt = 1, last_completion = 1 where component = "
+              f"'syscollector-packages'")
+    yield agent.id
+    injector.stop_receive()
+    query_wdb(f"global delete-agent {agent.id}")
+
+
+def test_min_full_scan_interval(configure_local_internal_options, get_configuration, configure_environment,
+                                restart_modulesd, add_simulated_agent):
+    """Check if the Vulnerability Detector module waits the minimal time set in "min_full_scan_interval"
+    to perform FULL_SCAN type scanning.
+
+    To do so, a simulated agent is added to the system, as soon as it is detected, the Vulnerability Detector
+    will launch an initial scan (BASELINE_SCAN) on it. Then the system date is changed to a time after
+    the date of the initial baseline scan, but before the minimum period set in "min_full_scan_interval",
+    so that the full scan should not be triggered. Finally, the date is changed again to a time when
+    the period set in "min_full_scan_interval" has expired, at which point the full scan should be triggered.
+
+    Args:
+        configure_local_internal_options (fixture): Set internal configuration for testing.
+        get_configuration (fixture): Get configurations from the module.
+        configure_environment (fixture): Configure a custom environment for testing.
+        restart_modulesd (fixture): Reset ossec.log and start a new monitor.
+        add_simulated_agent (fixture): Add a simulated agent to the manager for testing.
+    """
+    check_apply_test({'min_full_scan_interval'}, get_configuration['tags'])
+    config = get_configuration['metadata']
+    agent_id = add_simulated_agent
+
+    # Callbacks
+    callback_detect_full_scan_start = vd.make_vuln_callback(f"A full scan will be run on agent '{agent_id}'")
+    callback_detect_baseline_scan_start = vd.make_vuln_callback(f"A baseline scan will be run on agent '{agent_id}'")
+    callback_detect_scan_end = vd.make_vuln_callback(f"Finished vulnerability assessment for agent '{agent_id}'")
+
+    min_full_scan_interval = time_to_seconds(config['min_full_scan_interval'])
+    seconds_to_travel = min_full_scan_interval / 2
+
+    # Detect baseline scan.
+    wazuh_log_monitor.start(timeout=vd.VULN_DETECTOR_BASELINE_SCAN_TIMEOUT,
+                            callback=callback_detect_baseline_scan_start,
+                            error_message='No baseline scan detected in log.')
+
+    # Detect baseline scan completion.
+    wazuh_log_monitor.start(timeout=vd.VULN_DETECTOR_SCAN_TIMEOUT,
+                            callback=callback_detect_scan_end,
+                            error_message='No scan end detected in log.')
+
+    # Travelling in time, halfway through the period set in "min_full_scan_interval".
+    before = str(datetime.now())
```

Comment:
This string casting isn't necessary. The `f-string` from below already uses the `__str__` and the `__repr__` methods of the object.

Reply:
Done in: https://github.com/wazuh/wazuh-qa/pull/1369/commits/5045a8250c712c9081ddc6565b62c4c87b998a5c